argh_derive/src/lib.rs

```diff
@@ -196,8 +196,12 @@ impl<'a> StructField<'a> {
         Some(StructField { field, attrs, kind, optionality, ty_without_wrapper, name, long_name })
     }
 
-    pub(crate) fn arg_name(&self) -> String {
-        self.attrs.arg_name.as_ref().map(LitStr::value).unwrap_or_else(|| self.name.to_string())
+    pub(crate) fn positional_arg_name(&self) -> String {
+        self.attrs
+            .arg_name
+            .as_ref()
+            .map(LitStr::value)
+            .unwrap_or_else(|| self.name.to_string().trim_matches('_').to_owned())
```

Comment:
We might want `.to_string().to_kebab_case()` to match how we transform long names.

And what do you think about instead having something like

```rust
struct Foo {
    #[argh(positional, name = "foo")]
    foo_: Vec<String>,
}
```

That'd let users pick arbitrarily different names for the positional arguments?

Reply:
`to_kebab_case` would also turn `foo_bar` into `foo-bar` and it wasn't clear to me that we wanted that.

We could add the name argument too. I see that as a compliment rather than a replacement for this.

Reply:
Looks like `#[argh(positional, arg_name="foo")]` should work already.

Reply:
If that's the case, should we just trim this down into fixing the clippy lints, and close #149 with a wontfix and suggest using arg_name="foo"?

Reply:
Oh, it looks like `to_kebab_case()` automatically strips out any trailing `_` characters. Sure, this looks fine to me.

We could consider using to_kebab_case() later if this becomes significant.